niceml/data/datasets/dfdataset.py

```diff
@@ -121,24 +256,68 @@ def __len__(self):
     def on_epoch_end(self):
         """shuffles on epoch end"""
```

Comment:
Suggestion 'shuffles data on epoch end'

Reply:
The docstring was not correct or general enough for this function anyway. I have rewritten it.